server/src/main/java/com/defold/extender/builders/CSharpBuilder.java

```diff
@@ -197,7 +203,42 @@ private static ArrayList<String> makePathsAbsolute(String basePath, ArrayList<St
         return out;
     }
 
-    private static void addLibFlags(String platform, List<String> linkFlags) throws IOException {
+    private static void getExportsFlags(String platform, File buildDir, List<String> linkFlags) throws IOException {
+        // We need to add the exported symbols map
+        File exportsFile = new File(String.format("%s/defold_cs.exports", buildDir));
+
+        String exportsPattern = "-Wl,--version-script=%s";
+        if (ExtenderUtil.isMacOSTarget(platform) || ExtenderUtil.isIOSTarget(platform))
+        {
+            exportsPattern = "-exported_symbols_list \"%s\"";
+        }
+
+        // HACK: In anticipation of the fix for https://github.com/dotnet/runtime/issues/109341
+        // we have to make sure not all symbols are public, and at the same time respect the engine symbols
```

Comment:
Another wait for bug fixes. But in the meantime this should work for most cases. And should be relatively easy to update if we need to.